src/client/auth/oidc.rs

```diff
@@ -44,6 +45,7 @@ const DEFAULT_ALLOWED_HOSTS: &[&str] = &[
     "::1",
 ];
 
+/// State is a struct that contains the callback and cache for OIDC.
 #[derive(Clone)]
 pub struct State {
```

Comment:
unfortunately, this is needed for the `TypedBuilder` derive, but I did add a `new` function, and I  changed the other use of default to new.

Reply:
Ahh I had forgotten about the `TypedBuilder` requirement. I missed in the last PR that the field for this type in `Credential` was made non-optional; was that necessary for implementation/to avoid lots of unwraps? It'd be more consistent with other fields/types for `Credential.oidc_callback` to be an `Option<Callback>` (which would make the `Default` stuff easier since `Default` for `Option` is `None`), but not a huge deal if that doesn't work.

Reply:
Oh, I moved the Option inside the type. I suppose we could have options at several levels. I’m thinking the inner Arc could also be a Box

Reply:
Oh no, I remember now, it’s because of interior mutability requirements